### PR TITLE
Resolve supercollider - scala - scdoc conflicts

### PR DESCRIPTION
**Pull request by ranjithshegde**
Supercollider filtypes `*.sc` and `*.scd` conflict with Scala and scdoc respectively.
At the moment the plugins that implement supercollider (for either vim or neovim) simply override the filetypes which is not an elegant solution.

This PR is an attempt at fixing it with regex. It also adds the other supercollider filetype `*.quark`

EDIT: Apologies but I do not know how to add tests for regex detected filetypes in `testdir/test_filetype.vim`. Are there any guides or examples?

### CODE REVIEW

**Comment by ranjithshegde** [comment]
@brammool Apologies if this kind of tagging is rude here. 
Could use your advice on how to proceed. Some of the class library files have 10+ lines of comments before getting to the code. Which means expensive `getline(12)` checks, or other similar  for-loop functions that check first non-comment line (seen some examples in `/dist/ft.vim`)

Alternately I could follow other methods from filetype by defining a variable `g:supercollider_ft` or such, which would set `*.sc` and `*scd` to supercollider if exists. Seeing as scala and supercollider have nothing in common, the overlap of users who use both seems to be non-existent. So this might be the economical option. Which would you prefer?

**Comment by dkearns** [comment]
It's common to offer an override variable (`:help filetype-overrule`) and fallback to content detection.  E.g., https://github.com/vim/vim/blob/e08fde007308fffec818b4ad6eccf573f54bed5b/runtime/autoload/dist/ft.vim#L503-L513

**Comment by ranjithshegde** [comment]
Added a function to `dist/ft.vim`.

Should I also add extensions referred in that function to `test_filetype.vim` or does that apply to only non-regex detected filetypes?

**Comment by brammool** [comment]
Please also use a function for the *.scd detection.  It's just a few lines, but since the number of entries keeps growing it's good to keep filetype.vim as short as possible.

In the SClang() function, "setf scala" should be below "endfor".

The detection from the file contents definitely deserve a test.  You can find examples in the lower part of src/testdir/test_filetype.vim.

**Comment by brammool** [comment]
Unless you plan to do more, perhaps it's time to remove the "draft" label?

**Comment by ranjithshegde** [comment]
I dont understand why the tests are failing. Says related to popup and gui

File: src/testdir/test_filetype.vim [hunk 1]
```diff
@@ -1497,6 +1497,54 @@ func Test_prg_file()
   filetype off
 endfunc
 
+  " Test dist#ft#SClang()
```
Comment (zeertzjq) [inline src/testdir/test_filetype.vim]:
```suggestion
" Test dist#ft#SClang()
```

File: src/testdir/test_filetype.vim [hunk 2]
```diff
@@ -1497,6 +1497,54 @@ func Test_prg_file()
...
+  call writefile(['PingPong {', '//comment','*ar { arg'], 'srcfile.sc')
+  split srcfile.sc
+  call assert_equal('supercollider', &filetype)
+  bwipe!
+  call delete('srcfile.sc')
+
+  filetype off
+endfunc
+
+  " Test dist#ft#SCD()
```
Comment (zeertzjq) [inline src/testdir/test_filetype.vim]:
```suggestion
" Test dist#ft#SCD()
```

**Comment by ranjithshegde** [comment]
Anything I should do for the single failing CI?

**Comment by brammool** [comment]
Running so many tests in so many configurations it's no surprise some tests fail.  We do try to reduce flakiness, but it can be hard.

**Pull request closed without merging**